table/tables/tables.go

```diff
@@ -338,13 +340,23 @@ func (t *tableCommon) UpdateRecord(ctx sessionctx.Context, h int64, oldData, new
 		}
 	}
 	colSize := make(map[int64]int64)
+	encodedCol := make([]byte, 0, 16)
 	for id, col := range t.Cols() {
-		val := int64(len(newData[id].GetBytes()) - len(oldData[id].GetBytes()))
-		if val != 0 {
-			colSize[col.ID] = val
+		encodedCol = encodedCol[:0]
+		encodedCol, err = tablecodec.EncodeValue(sc, encodedCol, newData[id])
+		if err != nil {
+			continue
 		}
+		newLen := len(encodedCol) - 1
+		encodedCol = encodedCol[:0]
+		encodedCol, err = tablecodec.EncodeValue(sc, encodedCol, oldData[id])
```

Comment:
Maybe we don't need to encode it to know the size:

- For integer types, we can binary search the value since the encoded size increases with value.
- For decimal type, we can get the size using `precision` and `frac` as calculated in `func (d *MyDecimal) ToBin(precision, frac int)`.

Reply:
Yeah, we can do this, but I feel that the cost of this computing would be close to `EncodeValue` actually?